session/payment/session_balance.go

```diff
@@ -124,12 +136,43 @@ func (sb *SessionBalance) Start() error {
 }
 
 func (sb *SessionBalance) loadInitialPromiseState() (promise.StoredPromise, error) {
-	lastPromise, err := sb.promiseStorage.GetLastPromise(sb.issuer)
+	lastPromise, err := sb.promiseStorage.GetLastPromise(sb.issuerID)
 	if err != nil {
-		// if an error occurs when fetching the last promise, issue a new id
-		sb.sequenceID, err = sb.promiseStorage.GetNewSeqIDForIssuer(sb.issuer)
+		if err.Error() == errBoltNotFound.Error() {
+			// if not found, issue a new sequenceID
+			lastPromise.SequenceID, err = sb.promiseStorage.GetNewSeqIDForIssuer(sb.consumerID, sb.receiverID, sb.issuerID)
+			sb.sequenceID = lastPromise.SequenceID
+			return lastPromise, err
+		}
 		return lastPromise, err
 	}
+
+	/* If we're providing a service for multiple consumers under the same issuer,
+	we'll need to check if the consumer ID matches.
+
+	If it does not - we'll need to do a thorough lookup since there's a case here,
+	where we might have issued a tiny amount that the provider did not clear,
+	but did clear the following promises.
+
+	In this case we'll need to issue a new id.
+	If none of the promises were cleared, we can reuse the old one.
+
+	TODO: This could turn out to be expensive, and we might just want to skip this check and issue a new ID instead.
+	*/
```

Comment:
I suggest not to invest time too much here for now since our new payment model does this a bit differently with a hash instead of seq id. In such case we could have a separte hash for each consumer even if promise issuer is the same.

Reply:
Since it's already implemented might as well just leave it here till we get more details on how it will actually work.